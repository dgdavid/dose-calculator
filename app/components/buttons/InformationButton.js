import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import { formatDate } from '../../lib/utils';

const InformationButton = ({ date, onPress }) => {
  if (!date) {
    return null;
  }

  return (
    <TouchableOpacity
      onPress={onPress}
    >
      <View>
        <Text style={styles.info}>
          Calculated at {formatDate(date)}
        </Text>
        <Text style={styles.instructions}>Tap here to calculate again</Text>
      </View>
    </TouchableOpacity>
  );
};

// TODO: add proptypes

const styles = StyleSheet.create({
  info: {
    padding: 8,
    alignSelf: 'center',
  },

  instructions: {
    alignSelf: 'center',
    fontSize: 9,
  },
});

export default InformationButton;
