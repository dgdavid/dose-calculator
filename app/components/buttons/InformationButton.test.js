import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { DateTime } from 'luxon';

import InformationButton from './InformationButton';

test('renders correctly when date is given', () => {
  const date = DateTime.utc(2018, 1, 31, 17, 50);

  const tree = renderer.create(
    <InformationButton date={date} /> // eslint-disable-line
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('rendering nothing when date is not given', () => {
  const tree = renderer.create(<InformationButton />).toJSON();

  expect(tree).toMatchSnapshot();
});
