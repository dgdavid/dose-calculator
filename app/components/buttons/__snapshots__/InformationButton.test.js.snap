// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering nothing when date is not given 1`] = `null`;

exports[`renders correctly when date is given 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "padding": 8,
        }
      }
    >
      Calculated at 
      2018-1-31 17:50
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 9,
        }
      }
    >
      Tap here to calculate again
    </Text>
  </View>
</View>
`;
